cognite/neat/rules/exporter/_rules2dms.py

```diff
@@ -443,47 +444,23 @@ def as_api_view_property(property_: Property, space: str) -> MappedPropertyApply
         else:
             return None
 
-    def to_cdf(self, client: CogniteClient):
-        """Write the the data model to CDF.
-
-        Args:
-            client: Connected Cognite client.
-
-        """
-        existing_data_model = self.find_existing_data_model(client)
-        existing_containers = self.find_existing_containers(client)
-        existing_views = self.find_existing_views(client)
-
-        if existing_data_model or existing_containers or existing_views:
-            raise exceptions.DataModelOrItsComponentsAlreadyExist(
-                existing_data_model, existing_containers, existing_views
-            )
-
-        self.create_space(client)
-        self.create_containers(client)
-        self.create_views(client)
-        self.create_data_model(client)
-
-    def find_existing_data_model(self, client: CogniteClient) -> DataModelId | None:
-        """Checks if the data model exists in CDF.
+    def find_existing_spaces(self, client: CogniteClient) -> set:
+        """Checks if the spaces exist in CDF.
 
         Args:
             client: Cognite client.
 
         Returns:
-             True if the data model exists, False otherwise.
+            True if the containers exist, False otherwise.
         """
-        if model := client.data_modeling.data_models.retrieve((self.space, self.external_id, self.version)):
-            cdf_data_model = model.latest_version()
-            logging.warning(exceptions.DataModelAlreadyExist(self.external_id, self.version, self.space).message)
-            warnings.warn(
-                exceptions.DataModelAlreadyExist(self.external_id, self.version, self.space).message,
-                category=exceptions.DataModelAlreadyExist,
-                stacklevel=2,
-            )
 
-            return cdf_data_model.as_id()
-        return None
+        existing_spaces = set()
+
+        for space in self.spaces:
+            if client.data_modeling.spaces.retrieve(space):
+                existing_spaces.add(space)
+
+        return existing_spaces
```

Comment:
Lets not do a request per space.
```suggestion
        return set(client.data_modeling.spaces.retrieve(list(self.spaces)).as_ids())
```

Reply:
Note the `find_existing_containers` and `find_existing_views` can equally be made into a simple one liner as this function.

Reply:
Done